src/app-layout/visual-refresh/trigger-button.tsx

```diff
@@ -46,64 +64,229 @@ function TriggerButton(
+  };
+
+  useEffect(() => {
+    if (hasTooltip && tooltipValue) {
+      const close = () => {
+        setShowTooltip(false);
+        setSupressTooltip(false);
+      };
+
+      const shouldCloseTooltip = (event: PointerEvent) => {
+        if (event.target && containerRef && (containerRef.current as any)?.contains(event.target as HTMLElement)) {
+          return false;
+        }
+        return true;
+      };
+
+      const handlePointerDownEvent = (event: PointerEvent) => {
+        if (shouldCloseTooltip(event)) {
+          close();
+        }
+      };
+
+      const handleKeyDownEvent = (event: KeyboardEvent) => {
+        if (event.key === 'Escape') {
+          close();
+        }
+      };
+
+      const wrapperDiv = containerRef.current;
+      if (wrapperDiv) {
+        wrapperDiv.addEventListener('pointerdown', handlePointerDownEvent);
+        wrapperDiv.addEventListener('keydown', handleKeyDownEvent);
+
+        return () => {
+          wrapperDiv.removeEventListener('pointerdown', handlePointerDownEvent);
+          wrapperDiv.removeEventListener('keydown', handleKeyDownEvent);
+        };
+      }
+    }
+  }, [containerRef, hasTooltip, tooltipValue]);
+
+  useEffect(() => {
+    if (tooltipVisible) {
+      return registerTooltip(() => {
+        setShowTooltip(false);
+        setSupressTooltip(false);
+      });
+    }
+  }, [tooltipVisible]);
 
   return (
-    <div className={clsx(styles['trigger-wrapper'], !highContrastHeader && styles['remove-high-contrast-header'])}>
-      {isMobile ? (
-        <InternalButton
-          ariaExpanded={ariaExpanded}
-          ariaLabel={ariaLabel}
-          ariaControls={ariaControls}
-          className={className}
-          disabled={disabled}
-          ref={ref}
-          formAction="none"
-          iconName={iconName}
-          iconSvg={iconSvg}
-          badge={badge}
-          onClick={onClick}
-          variant="icon"
-          __nativeAttributes={{
-            'aria-haspopup': true,
-            ...(testId && {
-              'data-testid': testId,
-            }),
-          }}
-        />
-      ) : (
-        <>
-          <button
-            aria-expanded={ariaExpanded}
-            aria-controls={ariaControls}
-            aria-haspopup={true}
-            aria-label={ariaLabel}
-            aria-disabled={disabled}
+    <div
+      ref={containerRef}
+      {...(hasTooltip && {
+        onPointerEnter: () => handlePointerEnter(),
+        onPointerLeave: () => handleBlur(false),
+        onFocus: e => handleFocus(e as any),
+        onBlur: () => handleBlur(true),
+      })}
+      className={clsx(styles['trigger-wrapper'], {
+        [styles['remove-high-contrast-header']]: !highContrastHeader,
+        [styles['trigger-wrapper-tooltip-visible']]: tooltipVisible,
+      })}
+    >
+      <div className={clsx(styles['trigger-wrapper'], !highContrastHeader && styles['remove-high-contrast-header'])}>
+        {isMobile ? (
+          <InternalButton
```

Comment:
usage of InternalButton component might be unnecessary here as the toolbar variant does not do this